core/http-auth-aws/src/main/java/software/amazon/awssdk/http/auth/aws/internal/signer/DefaultAwsV4HttpSigner.java

```diff
@@ -206,11 +204,8 @@ private static CompletableFuture<AsyncSignedRequest> doSign(AsyncSignRequest<? e
         SdkHttpRequest.Builder requestBuilder = request.request().toBuilder();
 
         CompletableFuture<V4Context> futureV4Context =
-            checksummer.checksum(request.payload().orElse(null)).thenApply(
-                checksum -> {
-                    requestBuilder.putHeader(X_AMZ_CONTENT_SHA256, checksum);
-                    return requestSigner.sign(requestBuilder);
-                });
+            checksummer.checksum(request.payload().orElse(null), requestBuilder)
+                       .thenApply(__ -> requestSigner.sign(requestBuilder));
```

Comment:
Is there a better way to do this without having an unused arg?

Reply:
Not that I know of.

Reply:
@haydenbaker answered it below, it seems to be because the code will be touching the request in more than one place. Perhaps it's easiest to review once all PRs touching this feature are in place.